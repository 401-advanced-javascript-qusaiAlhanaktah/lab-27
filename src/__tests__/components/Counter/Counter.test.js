import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../../../components/Counter/Counter.js';

describe('<Counter />', ()=>{
    it('exists at the start of our application',()=>{
        let app = shallow(<Counter />);
        expect(app.find('h3').exists()).toBeTruthy();
    });
    it('increment the state on click',()=>{
        // let app = mount(<Counter />);
        // let button = app.find('buuton1');
        // console.log(button)
        // button.simulate('click');
        // expect(app.counter).toBe(1)
    });
    it('renders correctly', () => {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});