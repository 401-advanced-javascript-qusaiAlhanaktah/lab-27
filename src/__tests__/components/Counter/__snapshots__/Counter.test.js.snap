// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    className="button1"
    onClick={[Function]}
  >
    click for Increment
  </button>
  <button
    className="button2"
    onClick={[Function]}
  >
    click for Decrement
  </button>
</section>
`;
